components/camel-workday/src/main/java/org/apache/camel/component/workday/WorkdayComponent.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.workday;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.camel.Endpoint;
+import org.apache.camel.ResolveEndpointFailedException;
+import org.apache.camel.support.DefaultComponent;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.annotations.Component;
+import org.apache.commons.httpclient.HttpClient;
+import org.apache.commons.httpclient.HttpConnectionManager;
+import org.apache.commons.httpclient.MultiThreadedHttpConnectionManager;
+
+/**
+ * Represents the component that manages {@link WorkdayEndpoint}.
+ */
+@Component("workday")
+public class WorkdayComponent extends DefaultComponent {
+
+    public static final String RAAS_ENDPOINT_URL = "https://%s/ccx/service/customreport2/%s%s";
+
+    private HttpClient httpClient;
+
+    @Metadata(label="advanced")
+    private WorkdayConfiguration workdayConfiguration;
+
+    protected Endpoint createEndpoint(String uri, String remaining, Map<String, Object> parameters) throws Exception {
+        WorkdayConfiguration workdayConfiguration = new WorkdayConfiguration();
+
+        setProperties(workdayConfiguration, parameters);
```

Comment:
This shouldn't be done, we just need to setProperties on the endpoint, so the second one should be sufficient, take a look at the other components.

Reply:
Removed the first call, the endpoint was updated to handle the properties.